docs/using-hive.md

```diff
@@ -2,72 +2,231 @@
+
+In addition to the default OpenShift DNS support, Hive offers a DNS feature called Managed DNS. With Managed DNS, Hive can automatically create delegated zones for approved base domains. For example, if hive.example.com exists and is specified as your managed domain, you can specify a base domain of cluster1.hive.example.com on your `ClusterDeployment`, and Hive will create this zone for you, wait for it to resolve, and then proceed with installation.
+
+### Pull Secret
+
+OpenShift installation requires a pull secret obtained from try.openshift.com. You can specify an individual pull secret for each cluster Hive creates, or you can use a global pull secret that will be used by all of the clusters Hive creates.
 
 ```bash
-oc create secret generic global-pull-secret --from-file=.dockerconfigjson=<path>/config.json --type=kubernetes.io/dockerconfigjson --namespace hive
+oc create secret generic mycluster-pull-secret .dockerconfigjson=/path/to/pull-secret --type=kubernetes.io/dockerconfigjson --namespace hive
 ```
 
-Edit the HiveConfig to add global pull secret.
+```yaml
+apiVersion: v1
+data:
+  .dockerconfigjson: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-pull-secret
+  namespace: hive
+type: kubernetes.io/dockerconfigjson
+```
+
+When a global pull secret is defined in the `hive` namespace and a `ClusterDeployment`-specific pull secret is specified, the registry authentication in both secrets will be merged and used by the new OpenShift cluster.
+When a registry exists in both pull secrets, precedence will be given to the contents of the cluster-specific pull secret.
+
+The global pull secret must live in the `hive` namespace and is referenced in the `HiveConfig` (also in the `hive` namespace).
+
+```yaml
+apiVersion: v1
+data:
+  .dockerconfigjson: REDACTED
+kind: Secret
+metadata:
+  name: global-pull-secret
+  namespace: hive
+type: kubernetes.io/dockerconfigjson
+
+```
 
 ```bash
 oc edit hiveconfig hive
 ```
 
-The global pull secret name must be configured in the HiveConfig CR.
-
 ```yaml
 spec:
   globalPullSecret:
     name: global-pull-secret
 ```
 
-### Provisioning
+### OpenShift Version
 
-Hive supports two methods of specifying what version of OpenShift you wish to install. Most commonly you can create a ClusterImageSet which references an OpenShift 4 release image.
+Hive needs to know what version of OpenShift to install. You can specify an individual OpenShift release image for each cluster Hive creates, or you can use a global `ClusterImageSet` that will be used by all of the clusters Hive creates.
```

Comment:
"will be used by all of the clusters Hive creates", this is inaccurate. ClusterImageSets are essentially for populating a dropdown, it's an indication that this release should be added to the index of versions that can be installed in this Hive instance. There can be many, and you can choose which one on your cluster deployment. If you don't set one, you can still use the specific release image ref.

Something like "can be referenced on any ClusterDeployment" or "can be used on any ClusterDeployment" perhaps.

Reply:
done

Reply:
(and thank you for helping me understand it)